### PR TITLE
fix(gossip): reject EpochSlots with null BitVec bits in sanitize

### PR DESCRIPTION
**Pull request by dymchenkko**
Implementations
- Null check in Uncompressed.sanitize rejects messages with null bits before they enter the gossip table
- Made BitVec.deinit null-safe while at it

Ramifications
- Peer sends one bad EpochSlots packet -> passes sanitize -> sits in table -> panics on eviction. Delayed crash makes it hard to trace back to the incoming message.

Tests
- Test now verifies sanitize rejects null bits and deinit survives them

Closes #1564

### CODE REVIEW

File: src/gossip/data.zig [hunk 1]
```diff
@@ -876,10 +876,12 @@ pub const Uncompressed = struct {
         if (self.num >= MAX_SLOT_PER_ENTRY) {
             return error.ValueOutOfBounds;
         }
+        if (self.slots.bits == null) {
+            return error.InvalidValue;
+        }
```
Comment (ajw221) [inline src/gossip/data.zig:881]: we want to cover the null path while also ensuring we don't drop any values that would otherwise be accepted in agave

```suggestion
        if (self.slots.bits == null and self.slots.len != 0) {
            return error.InvalidValue;
        }
```

File: src/gossip/data.zig [hunk 2]
```diff
@@ -2152,3 +2154,14 @@ test "sanitize vote" {
...
+test "BitVec null bits rejected by sanitize" {
+    const slot = Uncompressed{
+        .first_slot = 0,
+        .num = 0,
+        .slots = .{ .bits = null, .len = 0 },
+    };
+    try std.testing.expectError(error.InvalidValue, slot.sanitize());
+    // deinit is also safe to call with null bits
+    slot.deinit(std.testing.allocator);
+}
```
Comment (ajw221) [inline src/gossip/data.zig]: Continuing off of the previous suggested change, we should update the value in this test.

```suggestion
test "BitVec inconsistent null bits rejected by sanitize" {
    const slot = Uncompressed{
        .first_slot = 0,
        .num = 0,
        .slots = .{ .bits = null, .len = 8 },
    };
    try std.testing.expectError(error.InvalidValue, slot.sanitize());
    // deinit is also safe to call with null bits
    slot.deinit(std.testing.allocator);
}
```

`len = 8` here is on purpose and a valid length (multiple of 8) with no backing bits, so the `len % 8` check won't reject it and the only thing that can is the new `bits == null and len != 0` branch. If that branch gets removed or reordered, this value slips through `len % 8` and the test fails, which is what we want. A value like `len = 1` is malformed two ways, so `len % 8` would still reject it even if the null branch broke, and the test would keep passing and hide the regression.

This change swaps the old `len = 0` value for `len = 8`, so we lose coverage of the case the fix is about: `bits == null` with `len == 0` now being accepted. We should add a test asserting `sanitize()` succeeds on the empty value so that doesn't regress.

The best place for that is a test that bincode deserializes an empty `BitVec`, which also covers real data off the wire:

```zig
const wire = [_]u8{0} ** 25; // first_slot(8) + num(8) + bits=None(1) + len(8)
var stream = std.io.fixedBufferStream(&wire);
const decoded = try bincode.read(testing.allocator, Uncompressed, stream.reader(), .standard);
defer decoded.deinit(testing.allocator);
try testing.expect(decoded.slots.bits == null); // None -> null
try decoded.sanitize(); // empty is valid, matches agave
```

**Review by ajw221 — CHANGES_REQUESTED** [review]

File: src/gossip/data.zig (hunk 1 above)
Comment (ajw221) [reply src/gossip/data.zig:881]: [98a2451](https://github.com/Syndica/sig/commit/98a2451eb70c0513ceb9a8fd50270174d362caf1)

File: src/gossip/data.zig (hunk 2 above)
Comment (ajw221) [reply src/gossip/data.zig]: [98a2451](https://github.com/Syndica/sig/commit/98a2451eb70c0513ceb9a8fd50270174d362caf1)

File: src/gossip/data.zig (hunk 1 above)
Comment (dnut) [reply src/gossip/data.zig:881]: I just checked agave and I don't see that they are rejecting the item in sanitize when the slot bits are null

**Review by dnut — COMMENTED** [review]
Thanks for your contribution.

FYI we have a replacement for gossip in v2 and that's where we are focusing development effort going forward. Bug fixes are appreciated in v1 but not a priority unless it's actively causing us problems.

File: src/gossip/data.zig (hunk 1 above)
Comment (dnut) [reply src/gossip/data.zig:881]: @dymchenkko any thoughts? it's important for every node to replicate the exact same data, otherwise the network gets pull request noise
